### PR TITLE
Add tutorials with examples to Documentation

### PR DESCRIPTION
**Pull request by 9prady9**

### CODE REVIEW

**Comment by 9prady9** [comment]
@mlloreda @pavanky @jramapuram Feedback ?

File: doc/array_and_matrix_manipulation.md [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+* [flip()](#flip) - flip an array along a dimension
+* [join()](#join) - join up to 4 arrays
+* [reorder()](#reorder) - changes the dimension order within the array
+* [shift()](#shift) - shifts data along a dimension
+* [tile()](#tile) - repeats an array along a dimension
+* [transpose()](#transpose) - performs a matrix transpose
+
+Below we provide several examples of these functions and their use.
+
+### moddims()
```
Comment (jramapuram) [inline doc/array_and_matrix_manipulation.md]: Probably a stupid question: but isn't this already in the rust generated docs?

File: doc/configuring_arrayfire_environment.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
+% Configuring Arrayfire Environment
+
+Following are the list of environment and runtime configurations that will help enhance
+your experience with ArrayFire.
+
+# AF_PATH
```
Comment (jramapuram) [inline doc/configuring_arrayfire_environment.md]: This doc will definitely be helpful!

File: doc/external_docs.css [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/* unvisited link */
+a:link {
+    color: #4d76ae;
```
Comment (jramapuram) [inline doc/external_docs.css]: Is all this going into the gh-pages? I tried [your github.io for the fork](http://9prady.github.io/arrayfire-rust) but don't see anything.

File: src/array.rs [hunk 4]
```diff
@@ -175,6 +177,27 @@ impl Array {
...
+    /// Constructs a new Array object of specified dimensions and type
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use arrayfire::{Array, Dim4, DType};
+    /// let garbageVals = Array::new_empty(Dim4::new(&[3, 1, 1, 1]), DType::F32);
+    /// ```
+    #[allow(unused_mut)]
+    pub fn new_empty(dims: Dim4, aftype: DType) -> Array {
```
Comment (jramapuram) [inline src/array.rs]: shouldn't this go into another PR?

File: doc/array_and_matrix_manipulation.md (hunk 1 above)
Comment (9prady9) [inline doc/array_and_matrix_manipulation.md]: That is true, but having these collected set of examples in one location is probably a good place to start for beginners and they can dwell into parameter documentation and additional examples in the hyperlinked function documentation page - at least that is how I picture it in the long run. I do agree that more examples in each function doc page are to be added yet.

File: doc/external_docs.css (hunk 3 above)
Comment (9prady9) [inline doc/external_docs.css]: If you look at the `generate_docs.sh` that was added along with this change, that copies this new css file into the `target/doc` location while building the documentation.

File: src/array.rs (hunk 4 above)
Comment (9prady9) [inline src/array.rs]: Sure, why not. Let me cherry pick that into separate PR.

Comment (jramapuram) [inline src/array.rs]: It's not a big deal, just saying ;p

File: doc/array_and_matrix_manipulation.md (hunk 1 above)
Comment (jramapuram) [inline doc/array_and_matrix_manipulation.md]: Yea, good point. Probably helpful as a reference for people getting started (before they have their setup/etc)
